core/src/main/java/apoc/coll/Coll.java

```diff
@@ -50,6 +51,16 @@
 
     @Context public Transaction tx;
 
+    @UserFunction
+    @Description("apoc.coll.runningTotal(list1) - returns an accumulative array. For example apoc.coll.runningTotal([1,2,3.5]) return [1.0,3.0,6.5]")
+    public List<Double> runningTotal(@Name("list") List<Number> list) {
+        if (list == null || list.isEmpty()) return null;
+        AtomicDouble sum = new AtomicDouble();
+        return list.stream()
+                .map(i -> sum.addAndGet(i.doubleValue()))
```

Comment:
ottimo!!!

Aggiungi un controllo che ritorna il long se non c'è la parte decimale:
```suggestion
                .map(i -> {
                  double value = sum.addAndGet(i.doubleValue());
                  if (value == sum.longValue()) return sum.longValue();
                  return value;
                })
```

prova a cambiare il ritorno in `List<Number>`